crates/polars-time/src/replace.rs

```diff
@@ -0,0 +1,155 @@
+use polars_core::prelude::*;
+
+use crate::prelude::*;
+
+/// Replace specific time component of a `DatetimeChunked` with a specified value.
+#[cfg(feature = "dtype-datetime")]
+#[allow(clippy::too_many_arguments)]
+pub fn replace_datetime(
+    ca: &DatetimeChunked,
+    year: &Int32Chunked,
+    month: &Int8Chunked,
+    day: &Int8Chunked,
+    hour: &Int8Chunked,
+    minute: &Int8Chunked,
+    second: &Int8Chunked,
+    microsecond: &Int32Chunked,
+    ambiguous: &StringChunked,
+) -> PolarsResult<DatetimeChunked> {
+    let n = ca.len();
+
+    // For each argument, we must check if:
+    // 1. No value was supplied (None)       --> Use existing year from Series
+    // 2. Value was supplied and is a Scalar --> Create full Series of value
+    // 3. Value was supplied and is Series   --> Update all elements with the non-null values
+    let year = if year.len() == 1 {
+        // SAFETY: array has one value.
+        if let Some(value) = year.get(0) {
+            &Int32Chunked::full("".into(), value, n)
+        } else {
+            &ca.year()
+        }
+    } else {
+        &year.zip_with(&year.is_not_null(), &ca.year())?
+    };
+    let month = if month.len() == 1 {
```

Comment:
This file has got stray safety comments.

Reply:
Fixed.